src/XrdCl/XrdClZipArchive.cc

```diff
@@ -430,6 +430,19 @@ namespace XrdCl
                                         else
                                           std::tie( cdvec, cdmap ) = CDFH::Parse( buff, eocd->cdSize, eocd->nbCdRec );
                                         log->Dump( ZipMsg, "[0x%x] CD records parsed.", this );
+                                        uint64_t sumSize = 0;
+										uint64_t sumCompSize = 0;
+										for (auto it = cdvec.begin(); it != cdvec.end(); it++) {
+											sumSize += (*it)->uncompressedSize;
+											sumCompSize += (*it)->compressedSize;
+											if ((*it)->offset > archsize || (*it)->offset + (*it)->compressedSize > archsize
+													|| (*it)->offset + (*it)->uncompressedSize > archsize)
```

Comment:
Similarly as in the previous comment, you should only check the "compressed" size.